### PR TITLE
CAF-3876: Extracted required classes to a lib project

### PR DESCRIPTION
**Pull request by Anthony-McGreevy**

### CODE REVIEW

**Comment by Anthony-McGreevy** [comment]
Jira: https://jira.autonomy.com/browse/CAF-3876

File: worker-document-lib/src/main/resources/META-INF/services/com.hpe.caf.api.worker.WorkerFactoryProvider [hunk 1]
```diff
@@ -0,0 +1 @@
+com.hpe.caf.worker.document.DocumentWorkerFactoryProvider
```
Comment (mmaBel) [inline worker-document-lib/src/main/resources/META-INF/services/com.hpe.caf.api.worker.WorkerFactoryProvider]: Why is this here? The worker-document project has this file and DocumentWorkerFactoryProvider isn't in worker-document-lib.

**Review by mmaBel — CHANGES_REQUESTED** [review]
Update the main README with a description of the worker-document-lib project.

I'm not sold on lib being a great name, lib is very generic and could apply to worker-document itself. It's not much better but these are classes that serve utility purposes (allowing passing document without danger of it being modified) so worker-document-util?
